packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -22,12 +21,95 @@ export const Shadow = stagedComponent((props: ShadowProps) => {
       }
     }
 
-    const childWithKeyShadow = React.cloneElement(child, mergeProps(child.props, { style: shadowTokenStyleSet.key }));
+    const { style: childStyle = {}, ...restOfChildProps } = child.props;
+
+    const shadowViewStyleProps = getStylePropsForShadowViews(childStyle, shadowTokenStyleSet);
+    const innerShadowViewProps = mergeProps(restOfChildProps, shadowViewStyleProps.inner);
+    const outerShadowViewProps = mergeProps(final, shadowViewStyleProps.outer);
+
+    const childWithInnerShadow = React.cloneElement(child, innerShadowViewProps);
 
-    return <View {...mergedProps}>{childWithKeyShadow}</View>;
+    return <View {...outerShadowViewProps}>{childWithInnerShadow}</View>;
   };
 });
 
+const getStylePropsForShadowViews = memoize(getStylePropsForShadowViewsWorker);
+function getStylePropsForShadowViewsWorker(childStyleProps: any, shadowTokenStyleSet: any) {
+  const {
+    borderBottomWidth,
+    borderEndWidth,
+    borderLeftWidth,
+    borderRightWidth,
+    borderStartWidth,
+    borderTopWidth,
+    borderWidth,
+
+    margin,
+    marginBottom,
+    marginEnd,
+    marginHorizontal,
+    marginLeft,
+    marginRight,
+    marginStart,
+    marginTop,
+    marginVertical,
+
+    padding,
+    paddingBottom,
+    paddingEnd,
+    paddingHorizontal,
+    paddingLeft,
+    paddingRight,
+    paddingStart,
+    paddingTop,
+    paddingVertical,
+
+    ...restOfChildStyleProps
+  } = childStyleProps;
+
+  return {
+    inner: {
+      style: {
+        backgroundColor: 'red', // will not be shown, just needed in macOS/iOS to ensure buggy behaviour is suppressed
+        ...(padding && { padding }),
```

Comment:
I think you could just pass padding (and the other properties) directly through like

padding,
paddingBottom,
etc.

Is there a reason to need the spread operator + creating an object around every defined style property?

Reply:
+1. I feel like there should be a more generic way to do this.

Reply:
I had it originally like that (or something similar, had "padding: padding" - didn't know it could be just "padding") but I think either way will add a bunch of undefined style props - ex. if padding wasn't set on the child originally, now it'll have "padding: undefined, paddingBottom: undefined" etc. Changed it to this so that props are only added if they aren't undefined.

Reply:
imo that's fine since undefined is the same as it not being set at all. to RN it's the same.

Reply:
Okay, changed it back. I think it makes the output of the jest tests harder to read but maybe that's fine